workshop/nano1-vuex.md

```diff
@@ -0,0 +1,281 @@
+    {{ title }}
+  </li>
+</ul>
+```
+
+2. `components/Card.vue` This will present our game information in a card. It will receive a `name` and `image` properties.
+
+3. `data/data.json` Our mock-up games database, JSON format.
+
+## 2. Step 1 - Installing Vuex
+
+Since we are using codepen, the only thing we need to install Vuex is to add it to our list of 
+dependencies, so go to the `Dependencies` button on the bottom left of your screen and add it to
+our project.
+
+::: tip 💡
+When working on a local project, you should first add the dependency with `npm install vuex` or
+`yarn add vuex`
+:::
+
+Now that we have our dependency, lets add it to our application.
+
+First, create a new file on the project root `store.js`, this will hold our Vuex store.
+
+Open the file, and lets import Vue and Vuex:
+
+```js
+import Vue from "vue";
+import Vuex from "vuex";
+
+Vue.use(Vuex);
+```
+
+Dont forget to inject the dependency with `Vue.use(Vuex)`
+
+Let's now create our store. Add this to the file:
+
+```js
+const store = new Vuex.Store({
+  state: {
+    games: []
+  },
+
+  getters: {
+  },
+
+  mutations: {
+  },
+
+  actions: {
+  }
+});
+
+export default store;
+```
+
+We will cover everything in here in depth later on, but for now lets focus on `state`.
+
+Vuex is a state management library, and so we need to declare our global state for our application.
+Think of `state` on our store like a globally available `data() {}` in a Vue component. 
+
+We will add a `games` array which will hold our data soon.
+
+Go back to `main.js` and import our store:
+```js
+import store from "./store"; 
+```
+
+And finally add it to our Vue app:
+
+```js
+new Vue({
+  el: "#app",
+  components: { App },
+  template: "<App/>",
+  store
+}); 
+```
+
+## 3. Step 2 - Fetching our remote data
+
+In this nano we will pretend as if we were fetching data from some remote API, and receiving the contents of `data.json` into our application.
+
+You will notice that we are already importing it `import data from "./data/data.json";`, so head to our `getFromAPI()` method and add the following code:
+
+```js
+const apiData = data; // Our "pretend" ajax call
+apiData.forEach(game => {
+  this.$store.dispatch("addGame", game);
+}); 
+```
+
+We are assigning our imported `data` into a constant as a pretend ajax call, and then we are using the
+js array method `forEach` to loop through the array. 
+
+What exactly then is `$store.dispatch`?
+
+Vuex has three different state management types of methods.
+1. **Getters**. For getting data from the store
+2. **Mutations**. For mutating/changing the data in the store
+3. **Actions**. Similar to mutations, but they *commit* the mutations, and can hold other logic.
+
+In this case, we are using an **action** via *dispatch* to add a game to our store.
+
+::: tip 💡
+*Why am I not mutating the data with a commit?* It is a best practice to only commit from within actions.  
```

Comment:
I would disagree with this statement. Vuex docs doesn't have a strict recommendation for this as a best practice and Vue core team has different positions on this. While I personally support using only actions on component level, let's make the statement on `best practices` a bit softer: `it's considered as a one of good practices` for example. We can mention that it fact there is nothing bad in committing mutations directly when we have sync stuff but actions approach is more maintainable.